api/spec/v1.yaml

```diff
@@ -1073,18 +1123,92 @@ components:
       description: |
         A list of consensus layer accounts.
     
+    AddressPreimage:
+      description: |
+        The data from which a consensus-style address (`oasis1...`, aka ed25519-backed,
+        aka staking address) was derived. Notably, for EVM runtimes like Sapphire,
+        this links the oasis address and the Ethereum address.
+
+        Oasis addresses are derived from a piece of data, such as an ed25519
+        public key or an Ethereum address. The type of underlying data usually also
+        determines how the signatuers for this address are verified.
+        
+        Consensus supports only staking addresses (`context="oasis-core/address: staking"` below).
+        Runtimes support all types. This means that every consensus address is also
+        valid in every runtime. For example, in EVM runtimes, you can use staking
+        addresses, but only with oasis tools (e.g. a wallet); EVM contracts such as
+        ERC20 tokens or tools such as Metamask cannot interact with staking addresses.
+      type: object
+      properties:
+        context:
+          type: string
+          enum:
+            - "oasis-core/address: staking"
+            - "oasis-runtime-sdk/address: secp256k1eth"
+            - "oasis-runtime-sdk/address: sr25519"
+            - "oasis-runtime-sdk/address: multisig"
+            - "oasis-runtime-sdk/address: module"
+          description: |
+            The method by which the oasis address was derived from `address_data`.
+          example: "oasis-runtime-sdk/address: secp256k1eth"
+        context_version:
+          type: integer
+          nullable: true
+          default: 0
+          description: Version of the `context`.
```

Comment:
Nah Explorer would ignore it. It's for ... people who might want to do the derivation in the other direction themselves for some reason? IDK. It's here for completeness only. I imagine we will tag it with `json:omitempty`, and since eevrything is v0, the field will not explicitly appear on the wire anyway.
If it fills you with rage against bureaucracy and boilerplate, LMK and let's remove it.

Reply:
https://github.com/oasisprotocol/oasis-core/blob/v22.2.3/go/common/crypto/address/context.go#L43

the context is made of `identifier string, version uint8`. these two values get serialized into a binary string, and that binary string gets prepended to the data when hashing it into an address

https://github.com/oasisprotocol/oasis-core/blob/v22.2.3/go/common/crypto/address/address.go#LL99C21-L99C21

I think the front end ought to see a context (an identifier-version pair) such as oasis-runtime-sdk/address: secp256k1eth v0, recognize "hey that's the derivation from an ethereum compatible address," and display something like "from Ethereum-compatible address 0x..."

Reply:
I agree with everything you said @pro-wh, including the last sentence. But for that, the version is not needed. And it's slightly easier to parse just an enum-like context identifier, rather than a string that glues together the identifier and the version.
Anyway, no strong feelings here, I see, so resolving this and leaving as-is at least for now.